x-pack/legacy/plugins/reporting/export_types/printable_pdf/server/execute_job/index.ts

```diff
@@ -67,7 +73,36 @@ export const executeJobFactory: QueuedPdfExecutorFactory = async function execut
       })
     );
 
-    const stop$ = Rx.fromEventPattern(cancellationToken.on);
-    return process$.pipe(takeUntil(stop$)).toPromise();
+    const timeout$ = Rx.fromEventPattern(
+      resolve => {
+        timeoutId = setTimeout(async () => {
+          jobLogger.warn('PDF export timeout breach, attempting to capture page screenshot');
+          return browserDriverFactory
+            .screenshotPage()
```

Comment:
I like almost everything about this PR. I think it's a good idea to add this handler for the pre-timeout scenario.

I'm conflicted about it, because I have doubts that we can expect `browserDriverFactory.screenshotPage()` to actually work when we get to this point. In looking at [one possible root cause](https://github.com/puppeteer/puppeteer/issues/4290) where the screenshot call never resolves, this fix doesn't give us a way to recover from that.

At this point of the timeout scenario, is there anything else that can be done?
- Log info about the OS or the graphics capabilities
- Any extra debug logs printed out that we aren't already logging
- Log any suggestions to the user about anything they can try